Formula/lib/libunicode.rb

```diff
@@ -0,0 +1,67 @@
+class Libunicode < Formula
+  desc "Modern C++20 Unicode library"
+  homepage "https://github.com/contour-terminal/libunicode"
+  url "https://github.com/contour-terminal/libunicode/archive/refs/tags/v0.6.0.tar.gz"
+  sha256 "0c217f8264000f1b8c36e78969cb9cf91ac97de937cc141ab78e6b1ad7f404ef"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+
+  uses_from_macos "python" => :build
+
+  on_macos do
+    depends_on "llvm" if DevelopmentTools.clang_build_version <= 1500
+  end
+
+  on_linux do
+    depends_on "gcc" # C++20
+  end
```

Comment:
Note: if we plan to allow this on Linux, we **cannot** have other formula depend on this (at least on Linux), since the GCC dependency will propagate to dependents too.

Reply:
Also: what if the system GCC already provides C++20 support?

Ideally we'd rely entirely on the `fails_with` support for this and it'd intelligently handle the necessary compiler.

Reply:
It'll work fine with the system runtime libraries if they already provide C++20 support, and installing Homebrew GCC is not necessary.

However, `brew` isn't smart enough (yet?) to know that installing Homebrew GCC isn't needed, so it will install `gcc` as a dependency anyway. This is in part because `brew` has no way to tell if you're doing `depends_on "gcc"` because, say, you need `gfortran` or because you need a newer C++ standard. (It may be possible to infer this from `fails_with`, but we don't do that.)

Reply:
> > It may be possible to infer this from `fails_with`, but we don't do that.
> 
> Yeh, this is something I've wanted for maybe 10 years 😂 😭

Maybe we can interest @ZhongRuoyu or @cho-m into working on this :smile:

Reply:
considering the complexity of using gcc, I can just drop the linux part for now.

Reply:
This is a new formula, not a particularly high usage one, so I would suggest: drop archs where it does not build with system compiler. So no Linux, and no older macOS.

Reply:
yeah, that also works for me.